zone/perl_mob.cpp

```diff
@@ -6301,6 +6263,50 @@ XS(XS_Mob_AddNimbusEffect) {
 	XSRETURN_EMPTY;
 }
 
+XS(XS_Mob_ShieldAbility); /* prototype to pass -Wmissing-prototypes */
+XS(XS_Mob_ShieldAbility) {
+	dXSARGS;
+	if (items < 2 || items > 6)
+		Perl_croak(aTHX_ "Usage: Mob::ShieldAbility(THIS, uint32 target_id, [int32 max_shielder_distance = 15], [int32 shield_duration = 12000], [int32 shield_target_mitigation= 50], [int32 shielder_mitigation = 50]"); // @categories Spells and Disciplines
+	{
+		Mob *THIS;
+		uint32             target_id = (uint32)SvUV(ST(1));
+		int32			   max_shlder_distance;
+		int32			   shld_duration;
+		int32			   shld_target_mitigation;
+		int32			   shlder_mitigation;
+
+		VALIDATE_THIS_IS_MOB;
+		if (items < 3)
+			max_shlder_distance = 15;
+		else {
+			max_shlder_distance = max_shlder_distance = (int32)SvUV(ST(2));
```

Comment:
These elses can just be declared as defaults above

Reply:
Example

```cpp
XS(XS_Mob_ShieldAbility); /* prototype to pass -Wmissing-prototypes */
XS(XS_Mob_ShieldAbility) {
	dXSARGS;
	if (items < 2 || items > 6)
	Perl_croak(aTHX_ "Usage: Mob::ShieldAbility(THIS, uint32 target_id, [int32 max_shielder_distance = 15], [int32 shield_duration = 12000], [int32 shield_target_mitigation= 50], [int32 shielder_mitigation = 50]"); // @categories Spells and Disciplines
	{
		Mob    *THIS;
		uint32 target_id                = (uint32) SvUV(ST(1));
		int32  max_shielder_distance    = (int32) SvUV(ST(2));
		int32  shield_duration          = (int32) SvUV(ST(3));
		int32  shield_target_mitigation = (int32) SvUV(ST(4));
		int32  shielder_mitigation      = (int32) SvUV(ST(5));

		VALIDATE_THIS_IS_MOB;
		if (items < 3) {
			max_shlder_distance = 15;
		}

		if (items < 4) {
			shld_duration = 12000;
		}

		if (items < 5) {
			shld_target_mitigation = 50;
		}

		if (items < 6) {
			shlder_mitigation = 50;
		}

		THIS->ShieldAbility(target_id, max_shlder_distance, shld_duration, shld_target_mitigation, shlder_mitigation);

	}
	XSRETURN_EMPTY;
}
```

Reply:
Yours looks good, just replaced what I had with that.